lib/sensu-plugins-http/common.rb

```diff
@@ -0,0 +1,30 @@
+
+module Common
+  def initialize
+    super()
+    aws_config
+  end
+
+  def aws_config
+    Aws.config.update(
```

Comment:
it pulls a json config file from an s3 bucket and merges the configuration in to the current checks config object. It's more generic than just password or username needed to be used, so if an endpoint can't be exposed either then that can be pushed in there as well.